fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/MetricUnionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,233 @@
+        ]
+        availability2.getAllAvailableIntervals() >> [
+                (metricColumn1): Collections.emptyList()
+        ]
+
+        when:
+        metricUnionAvailability = new MetricUnionAvailability(physicalTables, Collections.singleton(metricColumn1))
+
+        then:
+        IllegalArgumentException exception = thrown()
+        exception.message.startsWith("While constructing MetricUnionAvailability, Metric columns are not unique")
+    }
+
+
+    def "getDataSourceNames returns sources from availabilities not from physical tables"() {
+        given:
+        availability1.getAllAvailableIntervals() >> Collections.emptyMap()
+        availability2.getAllAvailableIntervals() >> Collections.emptyMap()
+
+        metricUnionAvailability = new MetricUnionAvailability(physicalTables, Collections.emptySet())
+
+        PhysicalTable physicalTable3 = Mock(PhysicalTable)
+        physicalTable3.getAvailability() >> metricUnionAvailability
+        MetricUnionAvailability outerMetricUnionAvailability = new MetricUnionAvailability(
+                Collections.singleton(physicalTable3),
+                Collections.emptySet()
+        )
+
+        expect:
+        outerMetricUnionAvailability.getDataSourceNames() == [TableName.of('source1'), TableName.of('source2')] as Set
+    }
+
+    def "getAllAvailableIntervals returns the combined intervals of configured metric columns of all availabilities"() {
+        given:
+        availability1.getAllAvailableIntervals() >> [
+                (metricColumn1): [new Interval('2018-01-01/2018-02-01')]
+        ]
+        availability2.getAllAvailableIntervals() >> [
+                (metricColumn2): [new Interval('2019-01-01/2019-02-01')]
+        ]
+
+        metricUnionAvailability = new MetricUnionAvailability(physicalTables, Collections.singleton(metricColumn1))
+
+        expect:
+        metricUnionAvailability.getAllAvailableIntervals() == [
+                (metricColumn1): [new Interval('2018-01-01/2018-02-01')]
+        ]
+
+    }
+    def "getAvailableIntervals returns the intersection of requested columns when available intervals have #reason"() {
+        given:
+        availability1.getAllAvailableIntervals() >> [
+                (metricColumn1): Collections.emptyList()
+        ]
+        availability2.getAllAvailableIntervals() >> [
+                (metricColumn2): Collections.emptyList()
+        ]
+
+        availability1.getAvailableIntervals(_ as DataSourceConstraint) >> new SimplifiedIntervalList(
+                availableIntervals1.collect{it -> new Interval(it)} as Set
+        )
+        availability2.getAvailableIntervals(_ as DataSourceConstraint) >> new SimplifiedIntervalList(
+                availableIntervals2.collect{it -> new Interval(it)} as Set
+        )
+
+        schema1.getColumns(MetricColumn.class) >> Sets.newHashSet(metricColumn1)
+        schema2.getColumns(MetricColumn.class) >> Sets.newHashSet(metricColumn2)
```

Comment:
It would be good to include some dimensions in here to make sure it handles dimensions the way it should as well.